src/sagemaker/fw_utils.py

```diff
@@ -68,6 +68,27 @@ def create_image_uri(region, framework, instance_type, framework_version, py_ver
         .format(account, region, framework, tag)
 
 
+def validate_source_dir(script, directory):
+    """Validate that the source directory exists and it contains the user script
+
+    Args:
+        script (str):  Script filename.
+        directory (str): Directory containing the source file.
+
+    Raises:
+        ValueError: If ``directory`` does not exist, is not a directory, or does not contain ``script``.
+    """
+    if directory:
+        if not os.path.exists(directory):
+            raise ValueError('"{}" does not exist.'.format(directory))
+        if not os.path.isdir(directory):
+            raise ValueError('"{}" is not a directory.'.format(directory))
+        if script not in os.listdir(directory):
```

Comment:
This is a very small point - but doing a stat on os.path.join(directory, script) would be better, because it avoids listing the entire directory.

Reply:
I'd be tempted to rewrite this entire thing as:

```python
if not os.path.isfile(os.path.join(directory, script)): 
    raise ValueError('...')
```

Reply:
Good idea. This is was just taken from the original tar_and_upload_dir() but this is a great time to improve it 👍

Reply:
just realized I didnt do this. Let me do it now.